website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,350 @@
+
+##### Corrupted Block (Id: 3)
+
+This block type is never written to persistent storage. While reading a log block, if the block is corrupted, then the reader gets an instance of the Corrupted Block instead of a Data block. 
+
+##### Avro Block (Id: 4)
+
+Data block serializes the actual records written into the log file
+
+![spec_log_format_avro_block](/assets/images/spec/spec_log_format_avro_block.png)
+
+| Section        | #bytes   | Description                                                  |
+| -------------- | -------- | ------------------------------------------------------------ |
+| format version | 4        | version of the log file format                               |
+| record count   | 4        | total number of records in this block                        |
+| record length  | 8        | length of the record content to follow                       |
+| record content | variable | Row represented as an Avro record serialized using BinaryEncoder |
+
+##### HFile Block (Id: 5)
+
+The HFile data block serializes the records using the HFile file format. HFile datamodel is a key value pair and both are encoded as byte arrays. Hudi row key is encoded as avro string and the Avro record serialized using BinaryEncoder is stored as the value. HFile file format stores the rows in sorted order and with index to enable quick point reads and range scans. 
+
+##### Parquet Block (Id: 6)
+
+The Parquet Block serializes the records using the Apache Parquet file format. The serialization layout is similar to the Avro block except for the byte array content encoded as columnar Parquet format. This log block type enables efficient columnar scans and better compression. 
+
+> Different data block types offers different tradeoffs and picking the right block is based on the workload requirements and is critical for merge and read performance. 
+
+
+
+## Reader Expections
+
+Readers will use snapshot isolation to query a hudi dataset at a consistent point in time in the Hudi timeline.  The reader constructs the snapshot state using the following steps
+
+1. Pick an instant in the timeline (last successful commit or a specific commit version explicitly queried) and set that the commit time to compute the list of files to read from. 
+2. For the picked commit time, compute all the file slices that belongs to that specific commit time. For all the partition paths involved in the query, the file slices that belong to an successful commit before the picked commit should be included. The lookup on the filesystem could be slow and inefficient and can be further optimized by caching in memory or using the files (partition path to filename) index or with a support of a external timeline serving system.  
+3. For the merge on read table type, ensure the apprpriate merging rules are applied to apply the updates queued for the base in the log files.
+   1. [TODO] List merging rules - (ordering field).
```

Comment:
This really depends on the payload implementation. But, we can talk about default one maybe `OverwriteWithLatestAvroPayload`?